jenkins/version-def.sh

```diff
@@ -31,6 +31,7 @@ CUDA_CLASSIFIER=${CUDA_CLASSIFIER:-"cuda11"}
 PROJECT_VER=${PROJECT_VER:-"21.10.0-SNAPSHOT"}
 PROJECT_TEST_VER=${PROJECT_TEST_VER:-"21.10.0-SNAPSHOT"}
 SPARK_VER=${SPARK_VER:-"3.0.1"}
+SHUFFLE_SPARK_SHIM=${SHUFFLE_SPARK_SHIM:-"spark301"}
```

Comment:
If we want to try to keep this in sync automatically in case someone only sets `SPARK_VER` we could try something like this.  Should work for Apache Spark versions but probably doesn't do so well with others like EMR, Databricks, etc.  However it'd be nice to work for most common cases rather than requiring users to remember to set both all the time.
```suggestion
SHUFFLE_SPARK_SHIM=${SHUFFLE_SPARK_SHIM:-spark${SPARK_VER//./}}
```

Reply:
that is much better for the default value. I'll add a comment also.